ext/xmlreader/php_xmlreader.c

```diff
@@ -123,6 +123,33 @@ zval *xmlreader_get_property_ptr_ptr(zend_object *object, zend_string *name, int
 }
 /* }}} */
 
+static int xmlreader_has_property(zend_object *object, zend_string *name, int type, void **cache_slot)
+{
+	xmlreader_object *obj = php_xmlreader_fetch_object(object);
+	xmlreader_prop_handler *hnd = zend_hash_find_ptr(&xmlreader_prop_handlers, name);
+
+	if (hnd != NULL) {
+		if (type == ZEND_PROPERTY_EXISTS) {
+			return 1;
+		}
+
+		zval rv;
+		if (xmlreader_property_reader(obj, hnd, &rv) == FAILURE) {
```

Comment:
Note that you are reading a zval now, but you are never destroying it in the two return paths when this returns SUCCESS.

Reply:
Yeah, true... I have just fixed this :)